### DISCUSSION TITLE
How to make a visual indicator of an unsaved field?

### DISCUSSION THREAD

**Discussion by GabrielSieben** [filamentphp/filament#7009] [Help]
Hello,

I am using Filament Forms in a bit of an unusual way; and it would be a great help to me if there was a way where if the user changed the contents of a form, I could display a visual indicator (using JavaScript, not some server-back-and-forth) that the form hasn't been saved, so click the Save button (which is connected with `wire:click()`). Maybe some red text appears if any unsaved changes exist?

Gabriel

**Comment by pboivin** [comment]
Hi @gjsman,

I'm not sure this is possible... any class or html attribute you dynamically add to a field on the page will be removed when a Livewire request is made and the form re-renders (e.g. with reactive fields).

An idea to explore would be wrapping the entire page component in a `div` with a `change` or `blur` event listener, and add a single indicator for unsaved changes. In this event listener, you can inspect the `event.target` and determine if it's a form field or not.
